daprdocs/content/en/developing-applications/building-blocks/workflow/howto-manage-workflow.md

```diff
@@ -58,6 +70,41 @@ To terminate your workflow with an ID `12345678`, run:
 POST http://localhost:3500/v1.0-alpha1/workflows/dapr/12345678/terminate
 ```
 
+### Raise an event
+
+Run the following for your workflow to create "wait for external event" tasks that:
```

Comment:
I still do not understand this. Is there an API to register for an event and an API to raise and event?
Can you provide a complete example of registering for an event X and then the workflow being triggered by event X happening? Reading this, it seems raise event is more like registered for an event. Is that correct, or am  I misunderstanding this?

Reply:
@RyanLettieri any suggestions for what is the best way to demonstrate the HTTP raise event call better?

Reply:
Hmm I don't think we can actually register an event through the HTTP api, only call an already registered event. Just to double check, @cgillum , is this correct?

Reply:
Still confused here. between sending and registering for events 
 
So there is
1)A  language SDK specific way to register for events with `WaitForExternalEvent`you mention above ^^

2) And also there is this API
// Raise an event (an incoming purchase order) that your workflow will wait for. This returns the item waiting to be purchased.
await daprClient.RaiseWorkflowEventAsync(orderId, workflowComponent, workflowName, input, cts);

3) And then there is an HTTP way to send an event, but then says below that this will "Subscribe to external events" and "Await those tasks to block execution until the event is received.", so that description is wrong then?

Reply:
Oh, sorry, I wasn't paying close enough attention to the context of your question. Yes, this description is not accurate - there's nothing about this API that will do any kind of subscribing or awaiting. Only the workflow itself can do that.

Here's my proposal for how to describe this API:

*For workflow components that support subscribing to external events, such as the Dapr Workflow engine, you can use the following "raise event" API to deliver a named event to a specific workflow instance.*

```http
POST http://localhost:3500/v1.0-alpha1/workflows/<workflowComponentName>/<instanceID>/raiseEvent/<eventName>
```

Reply:
Yes, this is better.